### ISSUE TITLE
Downsampled Segmentations remove segmented nuclei when saving the results

### ISSUE THREAD

**Issue by sophiamaedler** [MannLabs/scPortrait#4]
When segmenting the same with and without downsampling the resulting nuclear masks vary dramatically. 

| Nucleus | Cytosol |
|--------|--------|
| <img width="780" alt="Screenshot 2024-06-04 at 12 34 33" src="https://github.com/MannLabs/SPARCSspatial/assets/15019107/fb7e2e70-76d9-4988-92a9-aaf4a178a47f">| <img width="779" alt="Screenshot 2024-06-04 at 12 34 41" src="https://github.com/MannLabs/SPARCSspatial/assets/15019107/dd8a8ead-1e21-4a81-b056-33600d02b201">|
| <img width="785" alt="Screenshot 2024-06-04 at 12 40 20" src="https://github.com/MannLabs/SPARCSspatial/assets/15019107/c33f8eed-268a-42aa-a476-f0fd242c7ca3"> | <img width="780" alt="Screenshot 2024-06-04 at 12 40 27" src="https://github.com/MannLabs/SPARCSspatial/assets/15019107/1366d5ff-c830-4696-a932-ebb23f58680a"> | 

Intermediate results visualised during the segmentation process via debug = True indicate that the segmentation itself is producing similar results. This indicates that upscaling/saving the results in the downsampled method is the culprit.

Code to reproduce the issue: 
```
import os
import sys
import torch

from sparcscore.pipeline.project import Project
from sparcscore.pipeline.workflows import ShardedCytosolSegmentationDownsamplingCellpose, CytosolSegmentationCellpose, ShardedCytosolSegmentationCellpose, CytosolSegmentationDownsamplingCellpose
from sparcscore.pipeline.extraction import HDF5CellExtraction
from sparcscore.utils.vis import plot_segmentation_mask

test_seg = f"{base_dir}/{slide}/test_segmentations/"

if not os.path.isdir(test_seg):
    os.makedirs(test_seg)
    
project_location = f"{test_seg}/seg1"

project = Project(project_location,
            config_path= "/fs/pool/pool-mann-maedler-shared/Datasets/macrophage_screen/BMDMs/training_data/SPARCS_workflows/config/config_segmentation.yml",
            overwrite=True,
            debug=True,
            segmentation_f=CytosolSegmentationCellpose,
            extraction_f=HDF5CellExtraction,
            )

image = np.load("image.pkl", allow_pickle= True)
project.load_input_from_array(image)
project.segment()

fig = plot_segmentation_mask(project)
fig = plot_segmentation_mask(project, image_channel=1, mask_channel= 1)

project = Project(project_location,
            config_path= "/fs/pool/pool-mann-maedler-shared/Datasets/macrophage_screen/BMDMs/training_data/SPARCS_workflows/config/config_segmentation.yml",
            overwrite=True,
            debug=True,
            segmentation_f=CytosolSegmentationDownsamplingCellpose,
            extraction_f=HDF5CellExtraction,
            )

image = np.load("image.pkl", allow_pickle= True)
project.load_input_from_array(image)
project.segment()

fig = plot_segmentation_mask(project)
fig = plot_segmentation_mask(project, image_channel=1, mask_channel= 1)
```

[image.pkl.zip](https://github.com/user-attachments/files/15549925/image.pkl.zip)
